agent/taskresource/envFiles/envfile.go

```diff
@@ -0,0 +1,198 @@
+package envFiles
+
+import (
+	"sync"
+	"time"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	resourcestatus "github.com/aws/amazon-ecs-agent/agent/taskresource/status"
+	"github.com/aws/amazon-ecs-agent/agent/s3/factory"
+	"github.com/cihub/seelog"
+	"github.com/aws/amazon-ecs-agent/agent/s3"
+	"path/filepath"
+	"strings"
+	"github.com/pkg/errors"
+	"github.com/aws/amazon-ecs-agent/agent/utils/oswrapper"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
+	"github.com/aws/amazon-ecs-agent/agent/api/task/status"
+)
+
+const (
+	envFileDirPath = "envfiles"
+	envTempFilePrefix = "tmp_env"
+
+	s3DownloadTimeout = 30 * time.Second
+)
+
+// represents envfile as a task resource
+// these env files are retrieved from s3
+type EnvironmentFileResource struct {
+	cluster   string
+	taskARN   string
+	createdAt time.Time
+	region    string
+	resourceDir string // path to store env var files
+
+	// env file related attributes
+	envfileLocations []apicontainer.EnvironmentFile // list of env file urls
```

Comment:
evnfileSource might be more intuitive. With "location" I immediately thought of the dir we are storing it on the host.

Reply:
ah yeah good point, the name was from when i originally had it as a list of URLs but then i changed it but didn't change the actual name. will update